cmd/eksctl-anywhere/cmd/import_images.go

```diff
@@ -135,15 +144,22 @@ func (c ImportImagesCommand) Call(ctx context.Context) error {
 			password,
 		),
 		TmpArtifactsFolder: artifactsFolder,
-		FileImporter:       fetchFileRegistry(c.RegistryEndpoint, username, password, artifactsFolder, c.includePackages),
+		FileImporter:       fileRegistry,
 	}
 
 	return importArtifacts.Run(ctx)
 }
 
-func fetchFileRegistry(registryEndpoint, username, password, artifactsFolder string, includePackages bool) artifacts.FileImporter {
+// Temporary: Move to factory.go
+func fetchFileRegistry(registryEndpoint, username, password, artifactsFolder string, includePackages bool) (artifacts.FileImporter, error) {
 	if includePackages {
-		return oras.NewFileRegistryImporter(registryEndpoint, username, password, artifactsFolder)
+		registry, err := content.NewRegistry(content.RegistryOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("creating registry: %w", err)
+		}
+		memoryStore := content.NewMemory()
+		bundlePusher := oras.NewPusher(registry, memoryStore)
+		return oras.NewFileRegistryImporter(registryEndpoint, username, password, artifactsFolder, bundlePusher), nil
 	}
-	return &artifacts.Noop{}
+	return &artifacts.Noop{}, nil
```

Comment:
Can we invert this logic? It normally reads better and helps focus on the important bits when you keep the 'core logic' as unindented as possible (this relates to 'returning early instead of nesting deeply').

```go
	if !includePackages {
		return &artifacts.Noop{}, nil
	}

	registry, err := content.NewRegistry(content.RegistryOptions{})
	if err != nil {
		return nil, fmt.Errorf("creating registry: %w", err)
	}
	memoryStore := content.NewMemory()
	bundlePusher := oras.NewPusher(registry, memoryStore)
	return oras.NewFileRegistryImporter(registryEndpoint, username, password, artifactsFolder, bundlePusher), nil
```